Automattic/jetpack
# Fix subscribers messaging in post publish panel

Fixes https://github.com/Automattic/jetpack/issues/25317

#### Changes proposed in this Pull Request:

I propose to fix subscribers messaging in the post publish panel by saving the meta field when the post is changed back to draft after it was already published. Then, we disable subscribers' notification information in the Publish panel for posts already marked as published.

As the fix relies on the meta field that is saved when a post is saved as "draft" from "published", it will fix issues for all posts that are:

* currently "published" and will be changed to "draft" in the future
* are currently "draft", have never been "published" and will be changed to "published" and then "draft" in the future

Posts that were already "published" in the past, and have been changed to "draft" before this fix, will still show the message until they are "published" and become "draft" again.

#### Other information:


#### Jetpack product discussion



N/A

#### Does this pull request change what data or activity we track or use?



No

#### Testing instructions:






##### Preconditions (Jetpack):
* Jetpack is in online mode 
* ensure that the test site has at least one subscriber

##### Preconditions (Simple sites)

* download both diffs (D87321-code and D87352-code),upload them to the sandbox, and apply
* bring the PR to the WPCOM sandbox using a new process too:
```bin/jetpack-downloader test jetpack fix/subscribers-messaging-in-publish-panel```
* sandbox simple site and api
* ensure that the test site has a least one subscriber

##### Testing steps

1. Go to a WP Admin -> Posts
2. Open a form to add a new post
3. Fill it with any title and content
4. Click on "Publish"
5. Confirm that the "Subscribers" block is displayed in the Publish panel
6. Click "Switch to draft" and then approve with "OK"
7. Click on "Publish"
8. Confirm that the "Subscribers" block is not displayed anymore
